### DISCUSSION TITLE
Changes to build configurations in Conan Center CI

### DISCUSSION THREAD

**Discussion by jcar87** [conan-io/conan-center-index#16729] [Announcements]
# Changes to build configurations in Conan Center CI

As part of our efforts to support Conan 2.0 (see our [roadmap](https://github.com/conan-io/conan-center-index/blob/master/docs/v2_roadmap.md)), we will be making changes to the build configurations in our CI service when building binary packages for Conan 1.x.

As the first step, we will stop building packages for some configurations for which there is lower demand according to our access logs. In a future update, we will increase the configurations for binary packages for Conan 2.0. For an up-to-date list, please check our [documentation](https://github.com/conan-io/conan-center-index/blob/master/docs/supported_platforms_and_configurations.md).

### 📢  Starting **11 April 2023**, new pull requests will no longer be built or tested for the following configurations
⚠️  The changes announced here affect packages published for Conan 1.x. 

### Linux
* Deprecate the `libstdc++` `compiler.libcxx` setting, in favour of `libstdc++11`.
* Deprecate `gcc` version `8`

Most Linux distributions that have included a version of gcc greater than 5.1 use the new ABI by default, and we strongly encourage you to do so as well. Please refer to the documentation [here](https://docs.conan.io/1/howtos/manage_gcc_abi.html) for instructions on how to update this setting in your Conan profile.
As for the supported `gcc` versions, we have observed that the versions included in Ubuntu LTS releases have much greater demand than versions released in-between. 

### macOS
* Deprecate versions 11 and 12 of Apple-Clang

Our access logs indicate that users update their version of Xcode alongside new versions of macOS, and thus there is far less demand for the older Apple-Clang version from the Xcode 11 and 12 series. In a future update, we will start building and verifying Apple-Clang 14.

### Windows
* Deprecate compiler version 15 for Visual Studio (Visual Studio 2017)
* Deprecate  static runtime (`/MT`) on Visual Studio

Our access logs indicate declining demand for binaries for Visual Studio 2017, with much greater adoption for Visual Studio 2019. In a future update, we will start building and verifying packages with Visual Studio 2022.

In an earlier update, we dropped `MT` runtime support for shared builds on Windows. As a further step, we will no longer support runtime MT when building with the `shared=False` option variant, as our access logs indicate that this configuration is seldom requested. 

#### So what happens after April 11th in case I need binaries for some of those configurations?

As per our policy, all binaries that have been published to this date will continue to be served by Conan Center. However, any updates to recipes from that date will no longer be verified with the configurations that we are deprecating. You are, however, still able to build them locally from source (with the `--build missing` flag). Please have a look at the documentation: [how to build a package from source](https://docs.conan.io/1/reference/commands/consumer/install.html#build-options)

#### What happens in case I find a bug for a configuration which is no longer supported?

If you encounter problems with configurations that are not covered by our CI system, we encourage you to open a new pull request contributing a fix (see below). If you are unable to do so, please feel free to open an issue 
 [issue](https://github.com/conan-io/conan-center-index/issues/) describing your problem with as much detail as possible, and the community will be happy to help you. 

####  Can I open a new pull request fixing an error for those unsupported configurations?

Yes, but in order to facilitate the job of reviewers, we kindly request that you provide complete build logs, as well as your Conan profile settings, so that we can independently verify your contribution on our end.

**Comment by uilianries** [comment]
The pull request https://github.com/conan-io/conan-center-index/pull/16968 has been merged and dropped all configurations mentioned before.
